cmake/kokkos_compiler_id.cmake

```diff
@@ -42,12 +42,8 @@ IF(Kokkos_ENABLE_CUDA)
     # If launcher was found and nvcc_wrapper was not specified as
     # compiler and `CMAKE_CXX_COMPILIER_LAUNCHER` is not set, set to use launcher.
     # Will ensure CMAKE_CXX_COMPILER is replaced by nvcc_wrapper
-    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang
-       AND NOT (Kokkos_ENABLE_IMPL_NVHPC_AS_DEVICE_COMPILER AND KOKKOS_CXX_COMPILER_ID STREQUAL NVHPC))
+    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
       IF(CMAKE_CXX_COMPILER_LAUNCHER)
-        IF(KOKKOS_CXX_COMPILER_ID STREQUAL NVHPC)
-          MESSAGE(STATUS "Using nvc++ as device compiler requires Kokkos_ENABLE_IMPL_NVHPC_AS_DEVICE_COMPILER=ON!")
-        ENDIF()
         MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or clang++!")
```

Comment:
Feels like this message should be updated to address nvhpc explicitly.

Reply:
After this pull request `nvcc_wrapper` and `clang++` are the only device compilers supported. Thus, the error message is correct (but wasn't quite accurate before).

Reply:
My thinking was that if someone heard that nvhpc is able to compile cuda code before, but gets this message, it is not immediately clear from the language that only these two are supported. Maybe it's just me, but I would prefer explicitly stating that nvhpc is not supported.

Reply:
I think I am fine not doing that. In particular since NVIDIA would rather not want us to advertise that it once did kinda support CUDA.